opencga-catalog/src/main/java/org/opencb/opencga/catalog/monitor/executors/AzureBatchExecutor.java

```diff
@@ -0,0 +1,122 @@
+package org.opencb.opencga.catalog.monitor.executors;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.batch.BatchClient;
+import com.microsoft.azure.batch.auth.BatchSharedKeyCredentials;
+import com.microsoft.azure.batch.protocol.models.*;
+import org.opencb.opencga.core.config.Configuration;
+import org.opencb.opencga.core.models.Job;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Created by wasim on 17/12/18.
+ */
+public class AzureBatchExecutor implements BatchExecutor {
+    public static final String VARIANT_INDEX_JOB = "variant-index-job";
+    public static final String VARIANT_ANALYSIS_JOB = "variant-analysis-job";
+    private static Logger logger;
+    private String batchAccount;
+    private String batchKey;
+    private String batchUri;
+    private String batchServicePoolId;
+    private String dockerImageName;
+    private String dockerArgs;
+    private BatchClient batchClient;
+    private PoolInformation poolInformation;
+
+    public AzureBatchExecutor(Configuration configuration) {
+        logger = LoggerFactory.getLogger(AzureBatchExecutor.class);
+        populateOptions(configuration);
+        this.batchClient = createBatchClient();
+        this.poolInformation = new PoolInformation().withPoolId(batchServicePoolId);
+    }
+
+    public void submitAzureTask(Job job) throws IOException {
+        String jobId = getOrCreateAzureJob(job.getType());
+        TaskAddParameter taskToAdd = new TaskAddParameter();
+        taskToAdd.withId(job.getId()).withCommandLine(job.getCommandLine()).withContainerSettings(
+                new TaskContainerSettings().withImageName(dockerImageName)
+                        .withContainerRunOptions(dockerArgs));
+        taskToAdd.withId(job.getId()).withCommandLine(job.getCommandLine());
+        batchClient.taskOperations().createTask(jobId, taskToAdd);
+    }
+
+    private String getOrCreateAzureJob(Job.Type jobType) throws IOException {
+        String job = getAzureJobType(jobType);
+        PagedList<CloudJob> cloudJobs = batchClient.jobOperations().listJobs();
+        for (CloudJob cloudJob : cloudJobs) {
+            if (cloudJob.id().equals(job)) {
+                return job;
+            }
+        }
+        batchClient.jobOperations().createJob(job, this.poolInformation);
+        return job;
+    }
+
+    private String getAzureJobType(Job.Type jobType) {
+        return jobType == Job.Type.ANALYSIS ? VARIANT_ANALYSIS_JOB : VARIANT_INDEX_JOB;
+    }
+
+    private BatchClient createBatchClient() {
+        BatchSharedKeyCredentials cred = new BatchSharedKeyCredentials(batchUri,
+                batchAccount, batchKey);
+        return BatchClient.open(cred);
+    }
+
+    @Override
+    public void execute(Job job, String token) throws Exception {
+        submitAzureTask(job);
```

Comment:
Shouldn't this add the token to the command line?

Reply:
This already contains sid, I will check with pedro why he still needs --session-id